modules/efk-logging-collector-fluentd-v-rsyslog.adoc

```diff
@@ -0,0 +1,36 @@
+// Module included in the following assemblies:
+//
+// * logging/efk-logging-fluentd.adoc
+
+[id="efk-logging-collector-fluentd-v-rsyslog_{context}"]
+= About Fluentd and Rsyslog
+
+You can choose between Fluentd and Rsyslog as the logging collector. 
+
+Some of the noteable differences between the two collectors are:
+
+* No throttling in Rsyslog. With Fluentd, an administrator can reduce the rate at which the logs are read in by Fluentd before being processed. This feature is not currently available with Rsyslog.
+
+* No _built-in_ support for remote syslog forwarding in rsyslog. With Fluentd, you can use the `secure-forward` plug-in to send a copy of its logs to an external log aggregator. This feature is not currently available with Rsyslog.
++
+For Rsyslog, you can edit the Rsyslog configmap to add support for Syslog log forwarding using the *omfwd* module, see link:https://www.rsyslog.com/doc/v8-stable/configuration/modules/omfwd.html[]. To send logs to a different Rsyslog instance, you can the *omrelp* module, see link:https://www.rsyslog.com/doc/v8-stable/configuration/modules/omrelp.html[]
+
+* The Event Router is not supported for the Rsyslog log collector.
+
+* remote syslog
+
+When switching between the two collectors can result in:
+
+* A gap in messages between scale down of one collector and scale up of the other when reading from the tail.
+
+* Duplicate messages between scale down of one collector and scale up of the other when reading from the head.
++
+By default, Rsyslog reads from the head of the container log files, and from the start of the journal. Switching to Rsyslog 
```

Comment:
Seems only documented about the issues and solutions when changing from fluentd to rsyslog, no information about changing from rsyslog to fluentd.

Reply:
You will have the same problems going from rsyslog to fluentd.  Either you configure fluentd to read from the beginning of the journald and log files, which means a lot of duplicate records, or you configure fluentd to read from the tail of the journal and log files, which means missing records.

Reply:
@richm Is there a way in Openshift to set whether Fluentd reads from head or tail? I see there are Fluentd plugins to do this. 
In the past, it seems we had a `JOURNAL_READ_FROM_TAIL` environment variable to configure this.

Reply:
For container logs, use `CONTAINER_READ_FROM_HEAD=false` to read from the tail - by default it will read from the head.  For the journal, use `JOURNAL_READ_FROM_HEAD=true` to read from the head - by default it will read from the tail.